Sources/Algorithms/Trim.swift

```diff
@@ -0,0 +1,40 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension BidirectionalCollection {
+
+  /// Returns a `SubSequence` formed by discarding all elements at the start and end of the Collection
+  /// which satisfy the given predicate.
+  ///
+  /// e.g. `[2, 10, 11, 15, 20, 21, 100].trim(where: { $0.isMultiple(of: 2) })` == `[11, 15, 20, 21]`
+  ///
+  /// - parameters:
+  ///    - predicate:  A closure which determines if the element should be omitted from the resulting slice.
+  ///
+  @inlinable
+  public func trimmed(where predicate: (Element) throws -> Bool) rethrows -> SubSequence {
+    var sliceStart = startIndex
+    var sliceEnd = endIndex
+    // Consume elements from the front.
+    while sliceStart != sliceEnd, try predicate(self[sliceStart]) {
```

Comment:
Can you use existing stdlib methods here, instead of searching for the start and end manually?

Reply:
I can use `firstIndex(where:)` to find the start, but not `lastIndex(where:)` to find the end (since we actually want the index _after_ the last index to match the predicate; this would mean an extra call to `index(after:)`).

I've tried to improve the documentation, too. It might be worth adding a `String.trimmed()` function which defaults to trimming whitespace.